L1Trigger/L1TTrackMatch/test/L1TkMETAnalyser.cc

```diff
@@ -0,0 +1,271 @@
+#include <cmath>
+#include <iostream>
+#include <map>
+#include <string>
+#include <vector>
+
+#include "CommonTools/UtilAlgos/interface/TFileService.h"
+#include "DataFormats/Candidate/interface/Candidate.h"
+#include "DataFormats/HepMCCandidate/interface/GenParticle.h"
+#include "DataFormats/L1TCorrelator/interface/TkEtMiss.h"
+#include "DataFormats/L1TCorrelator/interface/TkEtMissFwd.h"
+#include "DataFormats/L1TrackTrigger/interface/TTTypes.h"
+#include "DataFormats/L1Trigger/interface/EtSum.h"
+#include "DataFormats/L1Trigger/interface/Vertex.h"
+#include "DataFormats/Phase2TrackerDigi/interface/Phase2TrackerDigi.h"
+#include "DataFormats/TrackerCommon/interface/TrackerTopology.h"
+#include "FWCore/Framework/interface/one/EDAnalyzer.h"
+#include "FWCore/Framework/interface/Event.h"
+#include "FWCore/Framework/interface/EventSetup.h"
+#include "FWCore/Framework/interface/MakerMacros.h"
+#include "FWCore/MessageLogger/interface/MessageLogger.h"
+#include "FWCore/ParameterSet/interface/ParameterSet.h"
+#include "FWCore/ServiceRegistry/interface/Service.h"
+#include "Geometry/Records/interface/TrackerDigiGeometryRecord.h"
+#include "L1Trigger/L1TTrackMatch/interface/L1TkEtMissEmuAlgo.h"
+#include "TEfficiency.h"
+#include "TGraphAsymmErrors.h"
+#include "TGraphErrors.h"
+#include "TH1F.h"
+#include "TH2F.h"
+#include "TPad.h"
+#include "TProfile.h"
+#include "TTree.h"
+
+using namespace std;
+
+class L1TkMETAnalyser : public edm::one::EDAnalyzer<> {
```

Comment:
The use of `TFileService` needs to be told to the framework with
```suggestion
class L1TkMETAnalyser : public edm::one::EDAnalyzer<edm::one::SharedResources> {
```
and calling
```cpp
usesResource(TFileService::kSharedResource);
```
in the constructor.